src/Uno.UI/UI/Xaml/Controls/PasswordBox/PasswordBox.cs

```diff
@@ -150,21 +156,99 @@ public bool IsPasswordRevealButtonEnabled
 
 		private void OnIsPasswordRevealButtonEnabledChanged(DependencyPropertyChangedEventArgs e)
 		{
-			_isButtonEnabled = IsPasswordRevealButtonEnabled;
+			CheckRevealModeForScope();
+			OnIsPasswordRevealButtonEnabledChangedPartial(e);
+		}
+
+		partial void OnIsPasswordRevealButtonEnabledChangedPartial(DependencyPropertyChangedEventArgs e);
+		#endregion
+
+		#region PasswordRevealMode DependencyProperty
+		public PasswordRevealMode PasswordRevealMode
+		{
+			get => (PasswordRevealMode)this.GetValue(PasswordRevealModeProperty);
+			set => this.SetValue(PasswordRevealModeProperty, value);
+		}
+
+		public static global::Windows.UI.Xaml.DependencyProperty PasswordRevealModeProperty { get; } =
+			DependencyProperty.Register(
+				nameof(PasswordRevealMode),
+				typeof(bool),
+				typeof(PasswordBox),
+				new FrameworkPropertyMetadata(
+					defaultValue: PasswordRevealMode.Peek,
+					propertyChangedCallback: (s, e) => ((PasswordBox)s)?.OnPasswordRevealModeChanged(e)
+				)
+			);
+
+		private void OnPasswordRevealModeChanged(DependencyPropertyChangedEventArgs e)
+		{
+			CheckRevealModeForScope();
+		}
 
-			if (IsPasswordRevealButtonEnabled)
+		private void CheckRevealModeForScope()
+		{
+			// Only use IsPasswordRevealButtonEnabled if it is set and PasswordRevealMode is not
+			if (UseIsPasswordEnabledProperty)
 			{
-				VisualStateManager.GoToState(this, TextBoxConstants.ButtonVisibleStateName, true);
-			}
-			else
+				SetPasswordScope(true);
```

Comment:
don't you need to actually check the value of `IsPasswordRevealButtonEnabled` here?

Reply:
No, because regardless of whether or not the reveal button should be displayed, the scope should be password until the reveal button is pressed, at which point the password should be displayed/hidden

Reply:
I should however be checking it when re-enabling the button in the `OnPasswordChanged`